static/src/javascripts/projects/common/utils/fastdom-promise.js

```diff
@@ -0,0 +1,34 @@
+define([
+    'common/utils/_',
+    'fastdom',
+    'Promise'
+], function (
+    _,
+    fastdom,
+    Promise
+) {
+    return {
+        read: function (fn, context) {
+            return new Promise(function (resolve, reject) {
+                fastdom.read(function () {
+                    if (_.isFunction(fn)) {
+                        resolve(fn.call(context));
+                    } else {
+                        reject();
+                    }
+                });
+            });
+        },
+        write: function (fn, context) {
```

Comment:
The structure of the code for `read` and `write` here is identical. You could probably write the code once as a utility function that takes a parameter for the `fastdom` fn name, which `read` and `write` here defer to.